images/converter/default.go

```diff
@@ -60,41 +58,95 @@ type ConvertHooks struct {
 
 // IndexConvertFuncWithHook is the convert func used by Convert with hook functions support.
 func IndexConvertFuncWithHook(layerConvertFunc ConvertFunc, docker2oci bool, platformMC platforms.MatchComparer, hooks ConvertHooks) ConvertFunc {
-	c := &defaultConverter{
+	return newDefaultWithOpts(convertOpts{
 		layerConvertFunc: layerConvertFunc,
 		docker2oci:       docker2oci,
 		platformMC:       platformMC,
-		diffIDMap:        make(map[digest.Digest]digest.Digest),
 		hooks:            hooks,
+	}).Convert
+}
+
+func NewDefault(opts ...Opt) (*DefaultConverter, error) {
+	var copts convertOpts
+	for _, o := range opts {
+		if err := o(&copts); err != nil {
+			return nil, err
+		}
+	}
+
+	return newDefaultWithOpts(copts), nil
+}
+
+func newDefaultWithOpts(copts convertOpts) *DefaultConverter {
+	if copts.platformMC == nil {
+		copts.platformMC = platforms.All
+	}
+
+	return &DefaultConverter{
+		convertOpts: copts,
 	}
-	return c.convert
 }
 
-type defaultConverter struct {
-	layerConvertFunc ConvertFunc
-	docker2oci       bool
-	platformMC       platforms.MatchComparer
-	diffIDMap        map[digest.Digest]digest.Digest // key: old diffID, value: new diffID
-	diffIDMapMu      sync.RWMutex
-	hooks            ConvertHooks
+type DefaultConverter struct {
```

Comment:
Why do we need make it public? Isn't it always wrapped by `ConverterFunc`?

Reply:
Make it public so that we can use `converter.NewDefault`, it provides better extensibility than `DefaultIndexConvertFunc`, for example allow specifying `converter.WithSingleflight`.